packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -386,3 +402,529 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
   @override
   SemanticsBuilderCallback? get semanticsBuilder => null;
 }
+
+/// An base class for building animated scrollbars.
+///
+/// To add a scrollbar to a [ScrollView], simply wrap the scroll view
+/// widget in a [RawScrollbar] widget.
+///
+/// {@template flutter.widgets.Scrollbar}
+/// A scrollbar indicates which portion of a [Scrollable] widget is actually
+/// visible.
+///
+/// By default, the thumb will fade in and out as the child scroll view widget
+/// scrolls. When [isAlwaysShown] is true, and the associated
+/// [ScrollController] is provided for [controller], the scrollbar thumb will
+/// remain visible without the fade animation.
+///
+/// Scrollbars are interactive and will use the [PrimaryScrollController] if
+/// a [controller] is not set. Scrollbar thumbs can be dragged along the main axis
+/// of the [ScrollView] to change the [ScrollPosition].
+/// {@endtemplate}
+///
+// TODO(Piinks): Add code sample
+///
+/// See also:
+///  * [Scrollbar], a scrollbar styled for Material Design.
+///  * [CupertinoScrollbar], an iOS style scrollbar.
+///  * [ListView], which display a linear, scrollable list of children.
+///  * [GridView], which display a 2 dimensional, scrollable array of children.
+class RawScrollbar extends StatefulWidget {
```

Comment:
Originally I had this as an abstract class, but after our last brainstorm session, we discussed how Scrollbars will be used by default for scrollables on desktop and web.
We can use this scrollbar for the widgets layer scroll configuration, while the other two scroll configurations (material and cupertino) have scrollbars to match.